runelite-client/src/main/java/net/runelite/client/plugins/tzhaartimers/TzhaarTimersPlugin.java

```diff
@@ -0,0 +1,297 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.tzhaartimers;
+
+import com.google.inject.Provides;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.Client;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import net.runelite.client.util.Text;
+import javax.inject.Inject;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Arrays;
+import static net.runelite.api.ItemID.FIRE_CAPE;
+import static net.runelite.api.ItemID.INFERNAL_CAPE;
+
+@PluginDescriptor(
+	name = "Tzhaar Timers",
+	description = "Display elapsed time in the Fight Caves and Inferno",
+	tags = {"inferno", "fight", "caves", "cape", "timer", "tzhaar"}
+)
+@Slf4j
+public class TzhaarTimersPlugin extends Plugin
+{
+	private static final String START_MESSAGE = "Wave: 1";
+	private static final String WAVE_MESSAGE = "Wave:";
+	private static final String DEFEATED_MESSAGE = "You have been defeated!";
+	private static final String INFERNO_COMPLETE_MESSAGE = "Your TzKal-Zuk kill count is:";
+	private static final String FIGHT_CAVES_COMPLETE_MESSAGE = "Your TzTok-Jad kill count is:";
+	private static final String INFERNO_PAUSED_MESSAGE = "The Inferno has been paused. You may now log out.";
+	private static final String FIGHT_CAVE_PAUSED_MESSAGE = "The Fight Cave has been paused. You may now log out.";
```

Comment:
You could precompile a regex for these since they are so similar.
```
Your (TzTok-Jad|TzKal-Zuk) kill count is:
The (Inferno|Fight Cave) has been paused. You may now log out.
```